bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/TemplatedTransformer.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2019 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.rewriter.impl;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Map;
+
+import com.adobe.acs.commons.properties.PropertyAggregatorService;
+import com.adobe.acs.commons.properties.util.TemplateReplacementUtil;
+import com.adobe.acs.commons.rewriter.ContentHandlerBasedTransformer;
+import com.adobe.granite.rest.Constants;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.rewriter.ProcessingComponentConfiguration;
+import org.apache.sling.rewriter.ProcessingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.AttributesImpl;
+
+public class TemplatedTransformer extends ContentHandlerBasedTransformer {
```

Comment:
If the JSON filter approach to applying the templated variables works, I wonder if we should just use that approach for all cases rather than basically coding two separate handlers.  Thoughts?

Reply:
The nice thing right now is having the separation between HTML and JSON transforming so users don't have to opt into both (though this could be added in the filter with a config). I think both are using the 'best practice' for each content type